packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -195,14 +189,121 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 		return entry.public;
 	}
 
-	private localUpdate(data: DatastoreMessageContent, _forceBroadcast: boolean): void {
-		const content = {
+	/**
+	 * The combined contents of all queued updates. Will be undefined when no messages are queued.
+	 */
+	private queuedData: GeneralDatastoreMessageContent | undefined;
+
+	/**
+	 * The time at which the presence data must be sent. When presence updates are submitted, this value is calculated
+	 * based on the allowable latency for the update. A timer will fire at this time, sending any queued message
+	 * immediately. Note that if updates come in with allowable latencies that require sending before this deadline, those
+	 * messages will be sent immediately. Therefore this value can be considered the time of the next scheduled presence
+	 * update.
+	 */
+	private sendMessageDeadline: number = 0;
+
+	private enqueueMessage(
+		data: GeneralDatastoreMessageContent,
+		options: LocalUpdateOptions,
+	): void {
+		const allowableUpdateLatency = options.allowableUpdateLatency ?? 0;
+		const forceBroadcast = options.forceBroadcast;
+
+		const now = Date.now();
+		const updateDeadline = now + allowableUpdateLatency;
+		if (this.queuedData === undefined) {
+			// No queued message, so set the deadline based on the current message's allowable latency
+			this.sendMessageDeadline = updateDeadline;
+		}
+
+		// This function-local "datastore" will hold the merged message data
+		const queueDatastore = this.queuedData ?? {};
+
+		// Iterate over the current message data; individual items are workspaces
+		for (const workspaceName of Object.keys(data)) {
+			const workspaceData = data[workspaceName];
+
+			// Initialize the merged data as the queued datastore entry for the workspace
+			// Since the key might not exist, create an empty object in that case. It will
+			// be set explicitly after the loop.
+			const mergedData = queueDatastore[workspaceName] ?? {};
+
+			// Iterate over each value manager and its data, merging it as needed
+			for (const valueManagerKey of Object.keys(workspaceData)) {
+				for (const [clientSessionId, value] of brandedObjectEntries(
+					workspaceData[valueManagerKey],
+				)) {
+					// TODO: Should any values be ignored here? E.g. values with ignoreUnmonitored?
+					mergedData[valueManagerKey] ??= {};
+					mergedData[valueManagerKey][clientSessionId] = mergeValueDirectory(
+						mergedData[valueManagerKey][clientSessionId],
+						value,
+						0, // TODO: what value should be passed here?
+					);
```

Comment:
0 is correct. This is a time shift for incoming updates. Nothing local needs a time shift.